### ISSUE TITLE
Bug: spacing around lists with parbox=false

### ISSUE THREAD

**Issue by ras52** [T-F-S/tcolorbox#233]
I appreciate the `parbox` option is documented as experimental, however I think there is a bug in how spacing around lists works.  The following code should illustrate the problem.   Within a `tcolorbox`, the list has no spacing above it.  If you comment out the `\begin{tcolorbox}` and `\end{tcolorbox}` lines, the expected spacing appears.

```latex
\documentclass{article}
\usepackage{tcolorbox}
\usepackage{lipsum}
\usepackage{enumitem}
\setlist{noitemsep}

\begin{document} 
\begin{tcolorbox}[parbox=false]
\lipsum[1]
\begin{itemize}
\item Foo
\item Bar
\end{itemize}
\lipsum[4]
\end{tcolorbox}
\end{document}
```

Although this is most apparent with the `noitemsep` option from the [`enumitem` package](https://www.ctan.org/pkg/enumitem), I don't think it only manifests there.  If you remove those two lines, you still see the spacing above the list is reduced when it is inside a `tcolorbox`.

**Comment by ras52** [comment]
I don't think this is the same as #232 as I'm not using a bleeding-edge version of LaTeX:
```
XeTeX 3.14159265-2.6-0.999992 (TeX Live 2020/Debian)

**Comment by muzimuzhi** [comment]
Can you share more version info (e.g., which version of LaTeX2e and tcolorbox do you use) as well as a screenshot showing what output you get?

**Comment by ras52** [comment]
Does this tell you everything you need to know?

```
richard@quoich:~$ xelatex --version
XeTeX 3.14159265-2.6-0.999992 (TeX Live 2020/Debian)
kpathsea version 6.3.2
Copyright 2020 SIL International, Jonathan Kew and Khaled Hosny.
There is NO warranty.  Redistribution of this software is
covered by the terms of both the XeTeX copyright and
the Lesser GNU General Public License.
For more information about these matters, see the file
named COPYING and the XeTeX source.
Primary author of XeTeX: Jonathan Kew.
Compiled with ICU version 67.1; using 67.1
Compiled with zlib version 1.2.11; using 1.2.11
Compiled with FreeType2 version 2.10.4; using 2.10.4
Compiled with Graphite2 version 1.3.14; using 1.3.14
Compiled with HarfBuzz version 2.7.4; using 2.7.4
Compiled with libpng version 1.6.37; using 1.6.37
Compiled with poppler version 0.68.0

richard@quoich:~$ locate tcolorbox.sty
/usr/share/texlive/texmf-dist/tex/latex/lwarp/lwarp-tcolorbox.sty
/usr/share/texlive/texmf-dist/tex/latex/tcolorbox/tcolorbox.sty

richard@quoich:~$ head -n5 /usr/share/texlive/texmf-dist/tex/latex/tcolorbox/tcolorbox.sty
%% The LaTeX package tcolorbox - version 4.42 (2020/10/09)
%% tcolorbox.sty: Text color boxes
%%
%% -------------------------------------------------------------------------------------------
%% Copyright (c) 2006-2020 by Prof. Dr. Dr. Thomas F. Sturm <thomas dot sturm at unibw dot de>
```

**Comment by ras52** [comment]
![Screenshot from 2023-06-18 22-27-11](https://github.com/T-F-S/tcolorbox/assets/1429812/bcbba127-2585-4125-95d9-98e3763ebecf)

**Comment by muzimuzhi** [comment]
Reproducible until v5.0.2, but disappeared since v5.1.0. From [changelog](https://github.com/T-F-S/tcolorbox/blob/master/doc/latex/tcolorbox/CHANGES.md#510---2022-06-22) for v5.1.0, I guess current issue is similar to #171.

Since `tcolorbox` maintains only a single latest version, what I could suggest is to manually upgrade to a newer version of `tcolorbox`.
![image](https://github.com/T-F-S/tcolorbox/assets/6376638/0d979fac-2dc7-4308-b2f5-202e3319e116)

![image](https://github.com/T-F-S/tcolorbox/assets/6376638/03d6add3-d9fb-4a0b-a577-01a06ecd8930)

<details><summary>Example used to produce screenshoots above</summary>
<p>

```tex
\documentclass[twocolumn]{article}
\usepackage{tcolorbox}
\usepackage{lipsum}

\def\test#1{
  \begin{tcolorbox}[#1, title={\texttt{\detokenize{#1}}}]
    \lipsum[1][1-3]
    \begin{itemize}
      \item Foo
      \item Bar
    \end{itemize}
    \lipsum[1][4-6]
  \end{tcolorbox}
}

\begin{document} 
\twocolumn[\fmtname, \fmtversion; tcolorbox \csname tcb@version\endcsname\par\medskip]

\test{parbox=false} \newpage
\test{parbox=true}

\end{document}
```

</p>
</details>

**Comment by muzimuzhi** [comment]
> From [changelog](https://github.com/T-F-S/tcolorbox/blob/master/doc/latex/tcolorbox/CHANGES.md#510---2022-06-22) for v5.1.0, I guess current issue is similar to #171.

I can confirm this is duplicate of #171, thus already fixed starting from v5.1.0.
![image](https://github.com/T-F-S/tcolorbox/assets/6376638/1e551bd2-43be-4054-bd05-014002a501c6)


<details><summary>Example used to produce screenshot above (revised)</summary>
<p>

```tex
\documentclass[twocolumn]{article}
\usepackage{tcolorbox}
\usepackage{lipsum}

\makeatletter
\def\test#1{
  \begin{tcolorbox}[#1, title={\texttt{\detokenize{#1}}}]
    \lipsum[1][1-3]
    \begin{itemize}
      \item Foo
      \item Bar
    \end{itemize}
    \lipsum[1][4-6]
  \end{tcolorbox}

  \medskip
  \vskip\dimexpr\pagegoal-\pagetotal-330pt\relax
  \begingroup
  \UseName{str_if_eq:nnTF}{#1}{parbox=false}
    {\tcb@parbox@use@false}{}%
  \begin{minipage}{.9\columnwidth}%
    \lipsum[1][1-3]
    \begin{itemize}
      \item Foo
      \item Bar
    \end{itemize}
    \lipsum[1][4-6]
  \end{minipage}
  \endgroup
}
\makeatother

\begin{document} 
\twocolumn[\fmtname, \fmtversion; tcolorbox \csname tcb@version\endcsname\par\medskip]

\test{parbox=false} \newpage
\test{parbox=true}

\end{document}
```

</p>
</details>

**Comment by ras52** [comment]
Okay.  Thanks for your help.  It sounds like this can be closed as fixed.
